# Comparing `tmp/pciSeq-0.0.51.dev2-py3-none-any.whl.zip` & `tmp/pciSeq-0.0.52.dev1-py3-none-any.whl.zip`

## zipinfo {}

```diff
@@ -1,49 +1,54 @@
-Zip file size: 355179 bytes, number of entries: 67
--rw-rw-rw-  2.0 fat     1977 b- defN 23-Jun-05 23:03 pciSeq/__init__.py
--rw-rw-rw-  2.0 fat     9272 b- defN 23-Jun-05 17:44 pciSeq/app.py
--rw-rw-rw-  2.0 fat     3333 b- defN 23-May-28 20:42 pciSeq/config.py
+Zip file size: 361109 bytes, number of entries: 72
+-rw-rw-rw-  2.0 fat     2135 b- defN 23-Jun-25 12:12 pciSeq/__init__.py
+-rw-rw-rw-  2.0 fat    10910 b- defN 23-Jun-25 12:17 pciSeq/app.py
+-rw-rw-rw-  2.0 fat     3367 b- defN 23-Jun-24 12:16 pciSeq/config.py
 -rw-rw-rw-  2.0 fat     3678 b- defN 22-Apr-07 23:15 pciSeq/make_plot.py
 -rw-rw-rw-  2.0 fat        0 b- defN 22-Jan-19 17:53 pciSeq/src/__init__.py
--rw-rw-rw-  2.0 fat       27 b- defN 23-Jun-06 15:35 pciSeq/src/_version.py
+-rw-rw-rw-  2.0 fat       27 b- defN 23-Jun-25 19:22 pciSeq/src/_version.py
 -rw-rw-rw-  2.0 fat        0 b- defN 22-Jan-19 17:53 pciSeq/src/cell_call/__init__.py
--rw-rw-rw-  2.0 fat    20857 b- defN 23-May-29 22:10 pciSeq/src/cell_call/datatypes.py
+-rw-rw-rw-  2.0 fat    20857 b- defN 23-Jun-25 12:05 pciSeq/src/cell_call/datatypes.py
 -rw-rw-rw-  2.0 fat     2995 b- defN 23-Apr-26 22:13 pciSeq/src/cell_call/log_config.py
--rw-rw-rw-  2.0 fat    16544 b- defN 23-May-29 22:10 pciSeq/src/cell_call/main.py
--rw-rw-rw-  2.0 fat     3104 b- defN 23-May-28 20:42 pciSeq/src/cell_call/summary.py
--rw-rw-rw-  2.0 fat     8937 b- defN 23-May-28 20:42 pciSeq/src/cell_call/utils.py
+-rw-rw-rw-  2.0 fat    19153 b- defN 23-Jun-25 12:05 pciSeq/src/cell_call/main.py
+-rw-rw-rw-  2.0 fat     3104 b- defN 23-Jun-25 12:05 pciSeq/src/cell_call/summary.py
+-rw-rw-rw-  2.0 fat     9194 b- defN 23-Jun-25 12:11 pciSeq/src/cell_call/utils.py
+-rw-rw-rw-  2.0 fat        0 b- defN 23-Jun-11 12:10 pciSeq/src/diagnostics/__init__.py
+-rw-rw-rw-  2.0 fat      299 b- defN 23-Jun-24 12:16 pciSeq/src/diagnostics/config.py
+-rw-rw-rw-  2.0 fat     3336 b- defN 23-Jun-25 12:05 pciSeq/src/diagnostics/diagnostics.py
+-rw-rw-rw-  2.0 fat      796 b- defN 23-Jun-25 12:11 pciSeq/src/diagnostics/launch_diagnostics.py
+-rw-rw-rw-  2.0 fat     7162 b- defN 23-Jun-25 12:11 pciSeq/src/diagnostics/utils.py
 -rw-rw-rw-  2.0 fat        0 b- defN 22-Jan-19 17:53 pciSeq/src/preprocess/__init__.py
--rw-rw-rw-  2.0 fat    11253 b- defN 23-May-28 20:42 pciSeq/src/preprocess/cell_borders.py
+-rw-rw-rw-  2.0 fat    11253 b- defN 23-Jun-25 12:05 pciSeq/src/preprocess/cell_borders.py
 -rw-rw-rw-  2.0 fat     6973 b- defN 23-Apr-26 22:13 pciSeq/src/preprocess/segmentation.py
--rw-rw-rw-  2.0 fat     4585 b- defN 23-May-29 22:10 pciSeq/src/preprocess/spot_labels.py
--rw-rw-rw-  2.0 fat       88 b- defN 23-May-28 20:42 pciSeq/src/preprocess/utils.py
+-rw-rw-rw-  2.0 fat     4585 b- defN 23-Jun-25 12:05 pciSeq/src/preprocess/spot_labels.py
+-rw-rw-rw-  2.0 fat       88 b- defN 23-Jun-24 12:16 pciSeq/src/preprocess/utils.py
 -rw-rw-rw-  2.0 fat        0 b- defN 22-Feb-20 14:12 pciSeq/src/viewer/__init__.py
--rw-rw-rw-  2.0 fat     2256 b- defN 23-May-28 20:42 pciSeq/src/viewer/run_flask.py
--rw-rw-rw-  2.0 fat     5007 b- defN 23-Jun-06 15:37 pciSeq/src/viewer/stage_image.py
--rw-rw-rw-  2.0 fat     7638 b- defN 23-May-28 20:42 pciSeq/src/viewer/utils.py
--rw-rw-rw-  2.0 fat    10895 b- defN 23-May-28 20:42 pciSeq/static/2D/index.html
+-rw-rw-rw-  2.0 fat     2256 b- defN 23-Jun-24 12:16 pciSeq/src/viewer/run_flask.py
+-rw-rw-rw-  2.0 fat     5007 b- defN 23-Jun-25 12:05 pciSeq/src/viewer/stage_image.py
+-rw-rw-rw-  2.0 fat     7768 b- defN 23-Jun-25 15:45 pciSeq/src/viewer/utils.py
+-rw-rw-rw-  2.0 fat    10895 b- defN 23-Jun-24 12:16 pciSeq/static/2D/index.html
 -rw-rw-rw-  2.0 fat     6506 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/genes_datatable.html
 -rw-rw-rw-  2.0 fat     9622 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/css/index.css
--rw-rw-rw-  2.0 fat     3135 b- defN 23-May-28 20:42 pciSeq/static/2D/viewer/css/progress.css
--rw-rw-rw-  2.0 fat     7435 b- defN 23-Jun-04 11:39 pciSeq/static/2D/viewer/js/classConfig.js
--rw-rw-rw-  2.0 fat     1512 b- defN 23-Jun-05 23:03 pciSeq/static/2D/viewer/js/config.js
+-rw-rw-rw-  2.0 fat     3135 b- defN 23-Jun-24 12:16 pciSeq/static/2D/viewer/css/progress.css
+-rw-rw-rw-  2.0 fat     7435 b- defN 23-Jun-25 12:05 pciSeq/static/2D/viewer/js/classConfig.js
+-rw-rw-rw-  2.0 fat     1474 b- defN 23-Jun-25 12:05 pciSeq/static/2D/viewer/js/config.js
 -rw-rw-rw-  2.0 fat     5589 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/customControl.js
--rw-rw-rw-  2.0 fat    22840 b- defN 23-Jun-04 12:15 pciSeq/static/2D/viewer/js/dapi.js
+-rw-rw-rw-  2.0 fat    22811 b- defN 23-Jun-25 12:05 pciSeq/static/2D/viewer/js/dapi.js
 -rw-rw-rw-  2.0 fat     5984 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/dataLoader.js
 -rw-rw-rw-  2.0 fat    17232 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/donut.js
 -rw-rw-rw-  2.0 fat     4106 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/dt.js
--rw-rw-rw-  2.0 fat     9835 b- defN 23-Jun-04 11:58 pciSeq/static/2D/viewer/js/glyphConfig.js
+-rw-rw-rw-  2.0 fat     9835 b- defN 23-Jun-24 12:16 pciSeq/static/2D/viewer/js/glyphConfig.js
 -rw-rw-rw-  2.0 fat     4389 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/glyphPaths.js
 -rw-rw-rw-  2.0 fat     7224 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/glyphs.js
--rw-rw-rw-  2.0 fat    12281 b- defN 23-May-28 20:42 pciSeq/static/2D/viewer/js/index.js
+-rw-rw-rw-  2.0 fat    12281 b- defN 23-Jun-24 12:16 pciSeq/static/2D/viewer/js/index.js
 -rw-rw-rw-  2.0 fat     1073 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/progress.js
 -rw-rw-rw-  2.0 fat     6367 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/stage_cells.js
--rw-rw-rw-  2.0 fat    12064 b- defN 23-May-28 20:42 pciSeq/static/2D/viewer/js/stage_glyphs.js
+-rw-rw-rw-  2.0 fat    12064 b- defN 23-Jun-24 12:16 pciSeq/static/2D/viewer/js/stage_glyphs.js
 -rw-rw-rw-  2.0 fat     9540 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/stage_markers.js
--rw-rw-rw-  2.0 fat     7534 b- defN 23-Jun-04 11:56 pciSeq/static/2D/viewer/js/stage_markers_patched.js
+-rw-rw-rw-  2.0 fat     7534 b- defN 23-Jun-25 12:05 pciSeq/static/2D/viewer/js/stage_markers_patched.js
 -rw-rw-rw-  2.0 fat    15523 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/stage_polygons.js
 -rw-rw-rw-  2.0 fat     4471 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/streaming-tsv-parser.js
 -rw-rw-rw-  2.0 fat     4692 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/viewerUtils.js
 -rw-rw-rw-  2.0 fat      995 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/css/L.Control.Layers.Tree.css
 -rw-rw-rw-  2.0 fat      421 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/css/Leaflet.Coordinates-0.1.3.css
 -rw-rw-rw-  2.0 fat   122544 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/css/bootstrap.min.css
 -rw-rw-rw-  2.0 fat     8669 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/css/index.css
@@ -56,14 +61,14 @@
 -rw-rw-rw-  2.0 fat    15374 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/js/leaflet-pip.js
 -rw-rw-rw-  2.0 fat     5927 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/js/leaflet.textpath.js
 -rw-rw-rw-  2.0 fat      265 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/js/preloader.js
 -rw-rw-rw-  2.0 fat     9097 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/MarkerContainer.js
 -rw-rw-rw-  2.0 fat     4434 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/bezier-easing.js
 -rw-rw-rw-  2.0 fat   613808 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/example.min.js
 -rw-rw-rw-  2.0 fat    91874 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/tools.min.js
--rw-rw-rw-  2.0 fat     1091 b- defN 23-Jun-06 15:40 pciSeq-0.0.51.dev2.dist-info/LICENCE
--rw-rw-rw-  2.0 fat     4095 b- defN 23-Jun-06 15:40 pciSeq-0.0.51.dev2.dist-info/METADATA
--rw-rw-rw-  2.0 fat       92 b- defN 23-Jun-06 15:40 pciSeq-0.0.51.dev2.dist-info/WHEEL
--rw-rw-rw-  2.0 fat       48 b- defN 23-Jun-06 15:40 pciSeq-0.0.51.dev2.dist-info/entry_points.txt
--rw-rw-rw-  2.0 fat        7 b- defN 23-Jun-06 15:40 pciSeq-0.0.51.dev2.dist-info/top_level.txt
-?rw-rw-r--  2.0 fat     6413 b- defN 23-Jun-06 15:40 pciSeq-0.0.51.dev2.dist-info/RECORD
-67 files, 1299996 bytes uncompressed, 344753 bytes compressed:  73.5%
+-rw-rw-rw-  2.0 fat     1091 b- defN 23-Jun-25 19:22 pciSeq-0.0.52.dev1.dist-info/LICENCE
+-rw-rw-rw-  2.0 fat     4116 b- defN 23-Jun-25 19:22 pciSeq-0.0.52.dev1.dist-info/METADATA
+-rw-rw-rw-  2.0 fat       92 b- defN 23-Jun-25 19:22 pciSeq-0.0.52.dev1.dist-info/WHEEL
+-rw-rw-rw-  2.0 fat       48 b- defN 23-Jun-25 19:22 pciSeq-0.0.52.dev1.dist-info/entry_points.txt
+-rw-rw-rw-  2.0 fat        7 b- defN 23-Jun-25 19:22 pciSeq-0.0.52.dev1.dist-info/top_level.txt
+?rw-rw-r--  2.0 fat     6872 b- defN 23-Jun-25 19:22 pciSeq-0.0.52.dev1.dist-info/RECORD
+72 files, 1316828 bytes uncompressed, 349947 bytes compressed:  73.4%
```

## zipnote {}

```diff
@@ -30,14 +30,29 @@
 
 Filename: pciSeq/src/cell_call/summary.py
 Comment: 
 
 Filename: pciSeq/src/cell_call/utils.py
 Comment: 
 
+Filename: pciSeq/src/diagnostics/__init__.py
+Comment: 
+
+Filename: pciSeq/src/diagnostics/config.py
+Comment: 
+
+Filename: pciSeq/src/diagnostics/diagnostics.py
+Comment: 
+
+Filename: pciSeq/src/diagnostics/launch_diagnostics.py
+Comment: 
+
+Filename: pciSeq/src/diagnostics/utils.py
+Comment: 
+
 Filename: pciSeq/src/preprocess/__init__.py
 Comment: 
 
 Filename: pciSeq/src/preprocess/cell_borders.py
 Comment: 
 
 Filename: pciSeq/src/preprocess/segmentation.py
@@ -177,26 +192,26 @@
 
 Filename: pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/example.min.js
 Comment: 
 
 Filename: pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/tools.min.js
 Comment: 
 
-Filename: pciSeq-0.0.51.dev2.dist-info/LICENCE
+Filename: pciSeq-0.0.52.dev1.dist-info/LICENCE
 Comment: 
 
-Filename: pciSeq-0.0.51.dev2.dist-info/METADATA
+Filename: pciSeq-0.0.52.dev1.dist-info/METADATA
 Comment: 
 
-Filename: pciSeq-0.0.51.dev2.dist-info/WHEEL
+Filename: pciSeq-0.0.52.dev1.dist-info/WHEEL
 Comment: 
 
-Filename: pciSeq-0.0.51.dev2.dist-info/entry_points.txt
+Filename: pciSeq-0.0.52.dev1.dist-info/entry_points.txt
 Comment: 
 
-Filename: pciSeq-0.0.51.dev2.dist-info/top_level.txt
+Filename: pciSeq-0.0.52.dev1.dist-info/top_level.txt
 Comment: 
 
-Filename: pciSeq-0.0.51.dev2.dist-info/RECORD
+Filename: pciSeq-0.0.52.dev1.dist-info/RECORD
 Comment: 
 
 Zip file comment:
```

## pciSeq/__init__.py

```diff
@@ -56,13 +56,14 @@
 import pciSeq.src.cell_call.utils as utils
 from pciSeq.src.cell_call.log_config import attach_to_log
 
 if check_libvips():
     from pciSeq.src.viewer.stage_image import tile_maker
 else:
     logger.warning('>>>> libvips is not installed. Please see https://www.libvips.org/install.html <<<<')
-    logger.warning('>>>> This is required if you want to use functions from this package to <<<<')
-    logger.warning('>>>> make a pyramid of tiles and have an image as background in the viewer. <<<<')
-
+    logger.warning('>>>> This is required only if you want to add a background image to the viewer. <<<<')
+    logger.warning('>>>> and visualise your results after cell typing. <<<<')
+    logger.warning('>>>> To do cell typing, libvips can be ignored, it is *not* necessary.  <<<<')
+    # logger.warning('>>>> LIBVIPS_ENABLED is %s.  <<<<' % check_libvips())
```

## pciSeq/app.py

```diff
@@ -1,18 +1,21 @@
 import os
 import pandas as pd
 import numpy as np
 import tempfile
 import pickle
+import redis
 from typing import Tuple
 from scipy.sparse import coo_matrix, save_npz, load_npz
 from pciSeq.src.cell_call.main import VarBayes
 from pciSeq.src.cell_call.utils import get_out_dir
 from pciSeq.src.preprocess.spot_labels import stage_data
 from pciSeq.src.preprocess.utils import get_img_shape
+from pciSeq.src.diagnostics.utils import redis_db
+from pciSeq.src.diagnostics.launch_diagnostics import launch_dashboard
 from pciSeq.src.viewer.run_flask import flask_app_start
 from pciSeq.src.viewer.utils import copy_viewer_code, make_config_js, make_classConfig_js
 from pciSeq import config
 from pciSeq.src.cell_call.log_config import attach_to_log, logger
 
 ROOT_DIR = os.path.dirname(os.path.realpath(__file__))
 
@@ -108,24 +111,29 @@
     except Exception as err:
         raise
 
     # 1. get the hyperparameters
     cfg = init(opts)
 
     # 2. validate inputs
-    validate(spots, coo, scRNAseq)
+    cfg = validate(spots, coo, scRNAseq, cfg)  # cfg is mutated here by adding the key: 'is_redis_running'
 
-    # 3. prepare the data
+    # 3. launch the diagnostics
+    if cfg['launch_diagnostics'] and cfg['is_redis_running']:
+        logger.info('Launching the diagnostics dashboard')
+        launch_dashboard()
+
+    # 4. prepare the data
     logger.info(' Preprocessing data')
     _cells, cellBoundaries, _spots = stage_data(spots, coo)
 
-    # 4. cell typing
+    # 5. cell typing
     cellData, geneData, varBayes = cell_type(_cells, _spots, scRNAseq, cfg)
 
-    # 5. save to the filesystem
+    # 6. save to the filesystem
     if (cfg['save_data'] and varBayes.has_converged) or cfg['launch_viewer']:
         write_data(cellData, geneData, cellBoundaries, varBayes, cfg)
 
     # 6. do the viewer if needed
     if cfg['launch_viewer']:
         [h, w] = get_img_shape(coo)
         dst = get_out_dir(cfg['output_path'])
@@ -177,22 +185,36 @@
         os.makedirs(debug_dir)
     pickle_dst = os.path.join(debug_dir, 'pciSeq.pickle')
     with open(pickle_dst, 'wb') as outf:
         pickle.dump(varBayes, outf)
         logger.info(' Saved at %s' % pickle_dst)
 
 
+def export_db_tables(out_dir, con):
+    tables = con.get_db_tables()
+    for table in tables:
+        export_db_table(table, out_dir, con)
+
+
+def export_db_table(table_name, out_dir, con):
+    df = con.from_redis(table_name)
+    fname = os.path.join(out_dir, table_name + '.csv')
+    df.to_csv(fname, index=False)
+    logger.info(' Saved at %s' % fname)
+
+
 def init(opts):
     """
     Reads the opts dict and if not None, it will override the default parameter value by
     the value that the dictionary key points to.
     If opts is None, then the default values as these specified in the config.py file
     are used without any change.
     """
     cfg = config.DEFAULT
+    cfg['is_redis_running'] = False  # this is set at the validate function below
     if opts is not None:
         default_items = set(cfg.keys())
         user_items = set(opts.keys())
         assert user_items.issubset(default_items), ('Options passed-in should be a dict with keys: %s ' % default_items)
         for item in opts.items():
             if isinstance(item[1], (int, float, list)) or isinstance(item[1](1), np.floating):
                 val = item[1]
@@ -201,23 +223,46 @@
             else:
                 raise TypeError("Only integers, floats and lists are allowed")
             cfg[item[0]] = val
             logger.info(' %s is set to %s' % (item[0], cfg[item[0]]))
     return cfg
 
 
-def validate(spots, coo, sc):
+def validate(spots, coo, sc, cfg):
     assert isinstance(spots, pd.DataFrame) and set(spots.columns) == {'Gene', 'x', 'y'}, \
         "Spots should be passed-in to the fit() method as a dataframe with columns ['Gene', 'x', 'y']"
 
     assert isinstance(coo, coo_matrix), 'The segmentation masks should be passed-in as a coo_matrix'
 
     if sc is not None:
         assert isinstance(sc, pd.DataFrame), "Single cell data should be passed-in to the fit() method as a dataframe"
 
+    cfg['is_redis_running'] = check_redis_server()
+    return cfg
+
+
+def confirm_prompt(question):
+    reply = None
+    while reply not in ("", "y", "n"):
+        reply = input(f"{question} (y/n): ").lower()
+    return (reply in ("", "y"))
+
+
+def check_redis_server():
+    logger.info("check_redis_server")
+    try:
+        redis_db()
+        return True
+    except (redis.exceptions.ConnectionError, ConnectionRefusedError):
+        logger.info("Redis ping failed!. Diagnostics will not be called unless redis is installed and the service is running")
+        return False
+        # logger.info("Redis ping failed!. Trying to install redis server")
+        # if confirm_prompt("Do you want to install redis server?"):
+        #     logger.info("...installing...")
+
 
 if __name__ == "__main__":
     # set up the logger
     attach_to_log()
 
     # read some demo data
     _iss_spots = pd.read_csv(os.path.join(ROOT_DIR, 'data', 'mouse', 'ca1', 'iss', 'spots.csv'))
```

## pciSeq/config.py

```diff
@@ -49,15 +49,17 @@
     # Boolean, if True the output will be saved as tsv files in a folder named 'pciSeq' in your system's temp dir.
     'save_data': False,
 
     # Set here where the results will be saved. If default then they will be saved at your system's temp folder
     'output_path': ['default'],
 
     # if true the viewer will be launched once convergence has been achieved
-    'launch_viewer': False,
+    'launch_viewer': True,
+
+    'launch_diagnostics': True,
 
     # Use either np.float16 or np.float32 to reduce memory usage. In most cases RAM consumption shouldnt
     # need more than 32Gb RAM. If you have a dataset from a full coronal mouse slice with a high number of
     # segmented cells (around 150,000) a gene panel of more than 250 genes and 100 or more different
     # cell types (aka clusters, aka classes) in the single cell data then you might need at least 64GB on
     # your machine. Changing the datatype to a float16 or float32 will help keeping RAM usage to a lower
     # level
```

## pciSeq/src/_version.py

```diff
@@ -1 +1 @@
-__version__ = '0.0.51.dev2'
+__version__ = '0.0.52.dev1'
```

## pciSeq/src/cell_call/main.py

```diff
@@ -1,35 +1,50 @@
 import numpy as np
 import pandas as pd
 import scipy.spatial as spatial
+import time
 import numpy_groupies as npg
 from pciSeq.src.cell_call.datatypes import Cells, Spots, Genes, SingleCell, CellType
 from pciSeq.src.cell_call.summary import collect_data
 import pciSeq.src.cell_call.utils as utils
+from pciSeq.src.diagnostics.utils import redis_db
 from pciSeq.src.cell_call.log_config import logger
 
 
 class VarBayes:
     def __init__(self, _cells_df, _spots_df, scRNAseq, config):
         self.config = config
+        self.redis_db = redis_db(flush=True)
         self.cells = Cells(_cells_df, config)
         self.spots = Spots(_spots_df, config)
         self.genes = Genes(self.spots)
         self.single_cell = SingleCell(scRNAseq, self.genes.gene_panel, self.config)
         self.cellTypes = CellType(self.single_cell)
         self.cells.class_names = self.single_cell.classes
         self.nC = self.cells.nC                         # number of cells
         self.nG = self.genes.nG                         # number of genes
         self.nK = self.cellTypes.nK                     # number of classes
         self.nS = self.spots.nS                         # number of spots
         self.nN = self.config['nNeighbors'] + 1         # number of closest nearby cells, candidates for being parent
                                                         # cell of any given spot. The last cell will be used for the
                                                         # misread spots. (ie cell at position nN is the background)
+        self.iter_num = None
         self.has_converged = False
 
+    def __getstate__(self):
+        # set here attributes to be excluded from serialisation (pickling)
+        # It makes the pickle filesize smaller but maybe this will have to
+        # change in the future.
+        # These two attributes take up a lot of space on the disk:
+        # _gamma_bar and _log_gamma_bar
+        # FYI: https://realpython.com/python-pickle-module/
+        attributes = self.__dict__.copy()
+        del attributes['redis_db']
+        return attributes
+
     def initialise(self):
         self.cellTypes.ini_prior()
         self.cells.classProb = np.tile(self.cellTypes.prior, (self.nC, 1))
         self.genes.init_eta(1, 1 / self.config['Inefficiency'])
         self.spots.parent_cell_id = self.spots.cells_nearby(self.cells)
         self.spots.parent_cell_prob = self.spots.ini_cellProb(self.spots.parent_cell_id, self.config)
         self.spots.gamma_bar = np.ones([self.nC, self.nG, self.nK]).astype(self.config['dtype'])
@@ -39,14 +54,15 @@
         p0 = None
         iss_df = None
         gene_df = None
         max_iter = self.config['max_iter']
 
         self.initialise()
         for i in range(max_iter):
+            self.iter_num = i
 
             # 1. For each cell, calc the expected gene counts
             self.geneCount_upd()
 
             # 2. calc expected gamma
             self.gamma_upd()
 
@@ -68,14 +84,19 @@
 
             self.has_converged, delta = utils.hasConverged(self.spots, p0, self.config['CellCallTolerance'])
             logger.info(' Iteration %d, mean prob change %f' % (i, delta))
 
             # replace p0 with the latest probabilities
             p0 = self.spots.parent_cell_prob
 
+            if self.config['is_redis_running']:
+                logger.info('start db publish')
+                self.redis_upd()
+                logger.info('end db publish')
+
             if self.has_converged:
                 # self.counts_within_radius(20)
                 iss_df, gene_df = collect_data(self.cells, self.spots, self.genes, self.single_cell)
                 break
 
             if i == max_iter-1:
                 logger.info(' Loop exhausted. Exiting with convergence status: %s' % self.has_converged)
@@ -354,13 +375,44 @@
         # assert np.all(temp.cell_label==out.index)
         df = pd.DataFrame(out, index=self.cells.centroid.index, columns=gene_names)
         df = pd.merge(temp, df, right_index=True, left_on='cell_label', how='left')
 
         # ignore the first row, it is the background
         return df.iloc[1:, :]
 
+    # -------------------------------------------------------------------- #
+    def redis_upd(self):
+        logger.info("redis start")
+        eta_bar_df = pd.DataFrame({
+            'gene_efficiency': self.genes.eta_bar,
+            'gene': self.genes.gene_panel
+        })
+        self.redis_db.to_redis(eta_bar_df, "gene_efficiency", iteration=self.iter_num,
+                               has_converged=self.has_converged, unix_time=time.time())
+        self.redis_db.publish(eta_bar_df, "gene_efficiency", iteration=self.iter_num,
+                               has_converged=self.has_converged, unix_time=time.time())
+
+        pi_bar_df = pd.DataFrame({
+            'weight': np.bincount(self.cells.classProb.argmax(axis=1))/np.bincount(self.cells.classProb.argmax(axis=1)).sum(),
+            'class': self.cellTypes.names
+        })
+        self.redis_db.to_redis(pi_bar_df, "cell_type_prior", iter_num=self.iter_num,
+                               has_converged=self.has_converged, unix_time=time.time())
+
+        idx=[]
+        for i, row in enumerate(self.cells.classProb[1:, :]):  # ignore the top row, it corresponds to the background, it is not an actual cell
+            idx.append(np.argmax(row))
+        counts = np.bincount(idx)
+        # prob = np.bincount(idx) / np.bincount(idx).sum()
+        df = pd.DataFrame({
+            'class_name': self.cellTypes.names,
+            'counts': counts
+        })
+        self.redis_db.to_redis(df, "cell_type_posterior", iter_num=self.iter_num, has_converged=self.has_converged, unix_time=time.time())
+        self.redis_db.publish(df, "cell_type_posterior", iteration=self.iter_num, has_converged=self.has_converged, unix_time=time.time())
+        logger.info("redis end")
```

## pciSeq/src/cell_call/utils.py

```diff
@@ -5,15 +5,16 @@
 from urllib.request import urlopen
 import numpy as np
 import pandas as pd
 import numexpr as ne
 # import numba as nb
 import os
 import glob
-import logging
+import subprocess
+from email.parser import BytesHeaderParser
 
 dir_path = os.path.dirname(os.path.realpath(__file__))
 # logger = logging.getLogger(__name__)
 
 
 # def read_image_objects(img_obj, cfg):
 #     meanCellRadius = np.mean(np.sqrt(img_obj.area / np.pi)) * 0.5
@@ -268,8 +269,14 @@
 def get_out_dir(path, sub_folder=''):
     if path[0] == 'default':
         out_dir = os.path.join(tempfile.gettempdir(), 'pciSeq')
     else:
         out_dir = os.path.join(path[0], sub_folder, 'pciSeq')
     if not os.path.exists(out_dir):
         os.makedirs(out_dir)
-    return out_dir
+    return out_dir
+
+
+def get_pciSeq_install_dir():
+    p = subprocess.run(['pip', 'show', 'pciSeq'], stdout=subprocess.PIPE)
+    h = BytesHeaderParser().parsebytes(p.stdout)
+    return os.path.join(h['Location'], 'pciSeq')
```

## pciSeq/src/viewer/utils.py

```diff
@@ -1,11 +1,10 @@
 import pandas as pd
 import numpy as np
-import subprocess
-from email.parser import BytesHeaderParser
+from pciSeq.src.cell_call.utils import get_pciSeq_install_dir
 import shutil
 import json
 import os
 import glob
 import csv
 from pciSeq.src.cell_call.log_config import logger
 
@@ -67,25 +66,31 @@
     config = os.path.join(dst, 'viewer', 'js', 'classConfig.js')
     with open(config, 'w') as data:
         data.write(str(config_str))
     logger.info(' classConfig saved at %s' % config)
 
 
 def copy_viewer_code(cfg, dst):
-    p = subprocess.run(['pip', 'show', 'pciSeq'], stdout=subprocess.PIPE)
-    h = BytesHeaderParser().parsebytes(p.stdout)
-    pciSeq_dir = os.path.join(h['Location'], 'pciSeq')
+    pciSeq_dir = get_pciSeq_install_dir()
     dim = '2D'
     src = os.path.join(pciSeq_dir, 'static', dim)
 
     shutil.copytree(src, dst, dirs_exist_ok=True)
     logger.info(' viewer code (%s) copied from %s to %s' % (dim, src, dst))
     return dst
 
 
+def _get_file(OUT_DIR, filepath, n, header_line):
+    [filename, ext] = os.path.basename(filepath).split('.')
+    file = os.path.join(OUT_DIR, filename + '_%d.%s' % (n, ext))
+    handle = open(file, "a")
+    handle.write(header_line)
+    return file, handle
+
+
 def splitter_mb(df, dir_path, mb_size):
     """ Splits a text file in (almost) equally sized parts on the disk. Assumes that there is a header in the first line
     :param filepath: The path of the text file to be broken up into smaller files
     :param mb_size: size in MB of each chunk
     :return:
     """
     # OUT_DIR = os.path.join(os.path.splitext(filepath)[0] + '_split')
```

## pciSeq/static/2D/viewer/js/config.js

### js-beautify {}

```diff
@@ -1,10 +1,10 @@
 // NOTES:
 // 1. roi is the image size in pixels. Leave x0 and y0 at zero and set x1 to the width and y1 to the height
-// 2. zoomLevels, leave this at 10
+// 2. zoomLevels: maximum zoom levels.
 // 3. tiles should point to the folder that keeps your pyramid of tiles. If you do not have that just
 //    change the link to a blind one (use an empty string or change the jpg extension for example).
 //    The viewer should work without the dapi background.
 //    If the tiles are stored locally, they should be stored under the project root dir and the link
 //    should be the path from the project root dir too.
 // 4. cellData, geneData, cellBoundaries:
 //    4.1: mediaLink paths are with respect to the location of 'streaming-tsv-parser.js'
@@ -16,15 +16,15 @@
     return {
         roi: {
             "x0": 0,
             "x1": 7602,
             "y0": 0,
             "y1": 5471
         },
-        zoomLevels: 10, // maximum zoom levels. Leave that at 10.
+        zoomLevels: 10,
         tiles: 'https://storage.googleapis.com/ca1-data/img/262144px/{z}/{y}/{x}.jpg',
         cellData: {
             mediaLink: '../../data/cellData.tsv',
             size: "2179232"
         },
         geneData: {
             mediaLink: '../../data/geneData.tsv',
```

## pciSeq/static/2D/viewer/js/dapi.js

### js-beautify {}

```diff
@@ -6,20 +6,29 @@
         roi = cfg.roi;
 
     var a = map_dims[0] / (roi.x1 - roi.x0),
         b = -map_dims[0] / (roi.x1 - roi.x0) * roi.x0,
         c = map_dims[1] / (roi.y1 - roi.y0),
         d = -map_dims[1] / (roi.y1 - roi.y0) * roi.y0;
 
-    // This transformation maps a point from the roi domain to the domain defined by [0,0] amd [img[0], img[1]].
+    // This transformation maps a point from the roi domain to the domain defined by [0,0] and [someScalarX*256*2^maxZoom, someScalarY*256*2^maxZoom].
+    // The scalar is there to make the bounding have the same shape as the roi.
     var t = new L.Transformation(a, b, c, d);
 
     // The transformation in this CRS maps the the top left corner to (0,0) and the bottom right to (256, 256)
+    // Leaflet thinks that the map is 256px-by-256px wide. These are the dimension of the tile at
+    // zoom = 0.
+    // Each side of the map however is 256 * 2 ** maxZoomLevel pixels wide.
+    // For maxZoomLevel = 10 for example the map is 262144px-262144px
+    // Hence we have to specify a factor of 256/262144 = 1/1024.
+    // in general the factor is 256 / (256 * 2 ** maxZoomLevel)
+    var a_x = 256 / (256 * 2 ** cfg.zoomLevels),
+        c_y = 256 / (256 * 2 ** cfg.zoomLevels)
     L.CRS.MySimple = L.extend({}, L.CRS.Simple, {
-        transformation: new L.Transformation(1 / 1024, 0, 1 / 1024, 0),
+        transformation: new L.Transformation(a_x, 0, c_y, 0),
     });
 
     var southWest = L.latLng(map_dims[1], map_dims[0]),
         northEast = L.latLng(0, 0),
         mapBounds = L.latLngBounds(southWest, northEast);
 
     map = L.map('mymap', {
@@ -302,36 +311,28 @@
         var myDiv = "<div class='tab-pane active fade in' id='map-content'>" +
             "<div class='container-fluid'>" +
             "<div class='col-sm-12'>" +
             "<div class='row'>" +
             "<div class='myTitle' id='dtTitle' style='margin-bottom:5px'> <h4>Highlighted Cell</h4>  " +
             " <img src='https://cdn2.iconfinder.com/data/icons/snipicons/500/pin-128.png' class='ribbon'/> " +
             "</div>" +
-            "</div>" +
-            "<div class='row' style='background-color: rgba(255, 255, 255, 0.0)'>" +
-            "<div class='chart-wrapper'>" +
-            // "<div class='chart-title' id='dtTitle'> </div>" +
-            "<div class='chart-stage'>" +
+            "<div class='row'>" +
             "<div class='col-sm-5'>" +
-            "<div class='chart-stage' style='background-color: rgba(255, 255, 255, 0.8)'>" +
-            "<table id='dtTable' class='display compact custom' data-page-length='5' width=100%></table>" +
+            "<div class='col-sm-12' style='background-color: darkgrey; padding-left: 0px'>" +
+            "<table id='dtTable' class='display compact custom' data-page-length='5' width=100%'></table>" +
             "</div>" +
             "</div>" +
             "<div class='col-sm-7'>" +
             "<div class='chart-stage' style='background-color: rgba(255, 255, 255, 0.0)'> " +
             "<div class='summary' id='pie'> " +
             "<svg width='300' height='180'></svg>" +
             "</div>" +
             "</div>" +
             "</div>" +
             "</div>" +
-            "</div>" +
-            "</div>" +
-            "</div>" +
-            "</div>" +
             "</div>";
 
         return myDiv
     }
```

## Comparing `pciSeq-0.0.51.dev2.dist-info/LICENCE` & `pciSeq-0.0.52.dev1.dist-info/LICENCE`

 * *Files identical despite different names*

## Comparing `pciSeq-0.0.51.dev2.dist-info/METADATA` & `pciSeq-0.0.52.dev1.dist-info/METADATA`

 * *Files 5% similar despite different names*

```diff
@@ -1,10 +1,10 @@
 Metadata-Version: 2.1
 Name: pciSeq
-Version: 0.0.51.dev2
+Version: 0.0.52.dev1
 Summary: Probabilistic cell typing for spatial transcriptomics
 Home-page: https://github.com/acycliq/pciSeq
 Author: Dimitris Nicoloutsopoulos
 Author-email: dimitris.nicoloutsopoulos@gmail.com
 License: BSD
 Classifier: Programming Language :: Python :: 3
 Classifier: License :: OSI Approved :: BSD License
@@ -18,14 +18,15 @@
 Requires-Dist: scikit-learn
 Requires-Dist: tqdm
 Requires-Dist: flask
 Requires-Dist: numexpr
 Requires-Dist: diplib
 Requires-Dist: pyvips
 Requires-Dist: natsort
+Requires-Dist: redis
 Provides-Extra: interactive
 Requires-Dist: matplotlib (>=2.2.0) ; extra == 'interactive'
 Requires-Dist: jupyter ; extra == 'interactive'
 
 # pciSeq: Probabilistic Cell typing by In situ Sequencing
 A Python package that implements the cell calling algorithm as described in [Qian, X., et al. Nature Methods (2020)](https://www.nature.com/articles/s41592-019-0631-4)
 <p align="center">
```

## Comparing `pciSeq-0.0.51.dev2.dist-info/RECORD` & `pciSeq-0.0.52.dev1.dist-info/RECORD`

 * *Files 15% similar despite different names*

```diff
@@ -1,36 +1,41 @@
-pciSeq/__init__.py,sha256=fC8GCB3HZo08yLVuoOye0WOASPl3XHRKPFMjeutqfsQ,1977
-pciSeq/app.py,sha256=T5-pSjHqLh344Lx2NoUbIFdGPCLWAimJGD6VXXw5mI4,9272
-pciSeq/config.py,sha256=M7iR2OAbptE5UldUa6t2cCiR9cyDwFEaTQbsKEe7yYY,3333
+pciSeq/__init__.py,sha256=mxqjE8smLqDQuwzheJ_iA3gfcyV7c4x8lHQPwSEQwJk,2135
+pciSeq/app.py,sha256=WTBF8N7hbVz6DE3yroUgXL23ophYgyQno_s5V-A0MF4,10910
+pciSeq/config.py,sha256=DFgt5LpDUBlLNQNecM-TfoTry3388LbdRU4jAxhxDHA,3367
 pciSeq/make_plot.py,sha256=2AkncrVVvytr2keK505oMC0LV1Txli9WpsSGpjmht_4,3678
 pciSeq/src/__init__.py,sha256=47DEQpj8HBSa-_TImW-5JCeuQeRkm5NMpJWZG3hSuFU,0
-pciSeq/src/_version.py,sha256=FXox9FoDVJbwsSnwAJdpRAmuqVnr-tV5u4GYY9A9WqU,27
+pciSeq/src/_version.py,sha256=vEMUYFCvX9K7MEOv0lzZkZe_YDlbY9sfs-8nG0BqNSY,27
 pciSeq/src/cell_call/__init__.py,sha256=47DEQpj8HBSa-_TImW-5JCeuQeRkm5NMpJWZG3hSuFU,0
 pciSeq/src/cell_call/datatypes.py,sha256=WxWGXn9rQ9pgvlBJtbl2RP2jIK4iEAKtw92AeyovNhk,20857
 pciSeq/src/cell_call/log_config.py,sha256=vnm9TxlMEVG_GLPfl21GDEWTafEk3B-drTLsVQMfFEE,2995
-pciSeq/src/cell_call/main.py,sha256=n7DqQSmZvNRzYOzY2CYA_TexZjnSHuL9rH3Hj6MOFZ8,16544
+pciSeq/src/cell_call/main.py,sha256=CzF_XwgTojRL1Bke-f_nspapm-x9qGhW1ryu3g9thu0,19153
 pciSeq/src/cell_call/summary.py,sha256=f2snLqJIeywtYuHrJLM-hRvifgqRU2u2aCKE_JrZJPs,3104
-pciSeq/src/cell_call/utils.py,sha256=sIrlkwdUgbfH7MmpOLw-Po4mTjsEWwNHHciUArHPN0A,8937
+pciSeq/src/cell_call/utils.py,sha256=M4KJrS82pIHuCoxg2ej6NE9fJAVCvJw90T7YmWI2T7o,9194
+pciSeq/src/diagnostics/__init__.py,sha256=47DEQpj8HBSa-_TImW-5JCeuQeRkm5NMpJWZG3hSuFU,0
+pciSeq/src/diagnostics/config.py,sha256=kfPzvlFdGN0QgdfECwVV8lujXUiyim4fuetJoU4ziBQ,299
+pciSeq/src/diagnostics/diagnostics.py,sha256=L8tBgykN-KktrUS8YdIP5doi8meVVUUZU2PvOcUt72g,3336
+pciSeq/src/diagnostics/launch_diagnostics.py,sha256=p2t1wDc48iWvpfEjUt_-J5iK37EXPMg2O12y-pCeE_w,796
+pciSeq/src/diagnostics/utils.py,sha256=W77vAs1id1T47y7Wgt8VE9xut71TzgcN5ixXKEFPBMw,7162
 pciSeq/src/preprocess/__init__.py,sha256=47DEQpj8HBSa-_TImW-5JCeuQeRkm5NMpJWZG3hSuFU,0
 pciSeq/src/preprocess/cell_borders.py,sha256=eTuFuexqGMFEV4upuTRInoaAi75NN46TIcvJ36TbZ4Q,11253
 pciSeq/src/preprocess/segmentation.py,sha256=9fc47EsxeFQnNtbJ0vm-RqATsBBMP73J5chdRFJocd4,6973
 pciSeq/src/preprocess/spot_labels.py,sha256=ukYTnp0xnu9J4-bKCTLEDhHAfaQNCvE7teEFyU0OLRk,4585
 pciSeq/src/preprocess/utils.py,sha256=7UB-fOKsSY_a4MVPs7E6BoNZcwITqm2N2JFMZqNYbj0,88
 pciSeq/src/viewer/__init__.py,sha256=47DEQpj8HBSa-_TImW-5JCeuQeRkm5NMpJWZG3hSuFU,0
 pciSeq/src/viewer/run_flask.py,sha256=9HS_RW-O9xddAeiApd4m_w2JJmbqUGZ5lhQ-z_HUeVA,2256
 pciSeq/src/viewer/stage_image.py,sha256=0zvdVk_R52T43ueSt3K3DgdMZPUyJNxDSyG0Bni7O78,5007
-pciSeq/src/viewer/utils.py,sha256=8TtwWtoqox6Cy0otCIs6d10xwWUcEQgBkgVhG3HWVOg,7638
+pciSeq/src/viewer/utils.py,sha256=5jcauyJ2CKbcFxN4TTLVLsVlufAdqaLUQKXuZt6r7EU,7768
 pciSeq/static/2D/index.html,sha256=9aMR5hOaXugJpSOt3ypUY24z6blsfOS83Cd9LHbAHa0,10895
 pciSeq/static/2D/viewer/genes_datatable.html,sha256=v10lTjkXYeAv0DVuaWHOB6XgG887k-_ABXAgobDV-tY,6506
 pciSeq/static/2D/viewer/css/index.css,sha256=IntuFc7kNAaNkLSL3m_9fpnH67i5KJpNWJ0OIW4TmD4,9622
 pciSeq/static/2D/viewer/css/progress.css,sha256=0MfjX9kIwD4wBeC39LBi9bCvu4ZkHwg_vtbuUiy4CuI,3135
 pciSeq/static/2D/viewer/js/classConfig.js,sha256=xhfWJHcU1bXipgJEokkMdUsrPDifPHpp_UHzW2qYopg,7435
-pciSeq/static/2D/viewer/js/config.js,sha256=AR2sAXyfDIPAWCKTeFV_gSie3W7RHI7JJq5Nvuwk-Y4,1512
+pciSeq/static/2D/viewer/js/config.js,sha256=1Eo75ruuB3USoaed3K8opHU3nh7y--Vi8UrkzNwQ4mE,1474
 pciSeq/static/2D/viewer/js/customControl.js,sha256=niQd9FT0z_Dkp7v523n8plj7kNRnRpvADcHOChbDskY,5589
-pciSeq/static/2D/viewer/js/dapi.js,sha256=hZwgaEhIfMARoBStZoz3bMyO8PetD9NJufboq3flN-A,22840
+pciSeq/static/2D/viewer/js/dapi.js,sha256=VgUvZ82pLg1w4k_nPuPP7oDDOCAKyRUUrfS3S5dtLlU,22811
 pciSeq/static/2D/viewer/js/dataLoader.js,sha256=iD3Kv9XNsSD2YyU3ilau5iaxAEQEX5VJnQ06VWyEP-g,5984
 pciSeq/static/2D/viewer/js/donut.js,sha256=dn9S3ZnNTFdEaiDU7ZadoKjVyvYp9LEJm82EvCe1ySY,17232
 pciSeq/static/2D/viewer/js/dt.js,sha256=eJpyKHNJjs1jf-GiTEa2_UsxMadO5eRWs2J0sJMhFms,4106
 pciSeq/static/2D/viewer/js/glyphConfig.js,sha256=23YX0DGc7XQXuJrUJDt59lref_TZPwUGF_Pr4h8c-tk,9835
 pciSeq/static/2D/viewer/js/glyphPaths.js,sha256=M3gwhPIRR17Q6cokLU754uovl5VwHQciduJKLlPhUXY,4389
 pciSeq/static/2D/viewer/js/glyphs.js,sha256=m6VFmUxNDJju9fXG9VzOKOnxip2mGbaBOyQKOY5_6zg,7224
 pciSeq/static/2D/viewer/js/index.js,sha256=IFMrGHYnC0fKzDgTfVlTrvLAqdNxlkjv_3_WNSAc7hw,12281
@@ -55,13 +60,13 @@
 pciSeq/static/2D/viewer/js/lib/js/leaflet-pip.js,sha256=QqsilU0SHTaKA_qNSv_jEDH53cWO6T_RdforfybnbKM,15374
 pciSeq/static/2D/viewer/js/lib/js/leaflet.textpath.js,sha256=i5cmMPqpqXShZz0TvXU3ZQUi2Y07SIbIpMYbhPE8gWE,5927
 pciSeq/static/2D/viewer/js/lib/js/preloader.js,sha256=eaFWRf9yT-Td4C2t01P8fDEZYszYTaGDgQ1vjevNIDI,265
 pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/MarkerContainer.js,sha256=Z8TADCqYEQr25OEF7wljsOVAOpx87UrjFPI2nvIty70,9097
 pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/bezier-easing.js,sha256=yMoRDqJiglg0M0eWM25WGA5zzLWuJpOyC3a8aGFx0jw,4434
 pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/example.min.js,sha256=bZEnK8XSnShrpLYtOS7Xq7wnp-nW_nMZh4aaDc1Gt6U,613808
 pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/tools.min.js,sha256=A027Ub-oWXvKbz2B3pgMFS5xApmnEHXz9GQh79YZZnU,91874
-pciSeq-0.0.51.dev2.dist-info/LICENCE,sha256=6kbiFSfobTZ7beWiKnHpN902HgBx-Jzgcme0SvKqhKY,1091
-pciSeq-0.0.51.dev2.dist-info/METADATA,sha256=qXNLkD66UeyUSUcP4yJnaz2Zq6Hh2AGbxBZnxD1ijHQ,4095
-pciSeq-0.0.51.dev2.dist-info/WHEEL,sha256=G16H4A3IeoQmnOrYV4ueZGKSjhipXx8zc8nu9FGlvMA,92
-pciSeq-0.0.51.dev2.dist-info/entry_points.txt,sha256=RGOkAd0S7rrTuEInj-jKu5B7630i4R3mK9v9vx6nqc8,48
-pciSeq-0.0.51.dev2.dist-info/top_level.txt,sha256=mSonMN0SaB51m5yLGm5hFE1Fthf-2U4sRKxSezY-JMk,7
-pciSeq-0.0.51.dev2.dist-info/RECORD,,
+pciSeq-0.0.52.dev1.dist-info/LICENCE,sha256=6kbiFSfobTZ7beWiKnHpN902HgBx-Jzgcme0SvKqhKY,1091
+pciSeq-0.0.52.dev1.dist-info/METADATA,sha256=pkUDKEAKRvq4B2_mPn0KHGVumXYatmuoXHViKA5ESrI,4116
+pciSeq-0.0.52.dev1.dist-info/WHEEL,sha256=G16H4A3IeoQmnOrYV4ueZGKSjhipXx8zc8nu9FGlvMA,92
+pciSeq-0.0.52.dev1.dist-info/entry_points.txt,sha256=RGOkAd0S7rrTuEInj-jKu5B7630i4R3mK9v9vx6nqc8,48
+pciSeq-0.0.52.dev1.dist-info/top_level.txt,sha256=mSonMN0SaB51m5yLGm5hFE1Fthf-2U4sRKxSezY-JMk,7
+pciSeq-0.0.52.dev1.dist-info/RECORD,,
```

